lib/time/time.go

```diff
@@ -211,14 +211,13 @@ func (d Duration) AttrNames() []string {
 
 // CompareSameType implements comparison of two Duration values. required by
 // starlark.Comparable interface.
-func (d Duration) CompareSameType(op syntax.Token, v starlark.Value, depth int) (bool, error) {
-	cmp := 0
+func (d Duration) ThreeWayCompareSameType(v starlark.Value, depth int) (int, error) {
 	if x, y := d, v.(Duration); x < y {
-		cmp = -1
+		return -1, nil
```

Comment:
For all integers you can use this trick:
        
{ return x > y - x < y, nil } // see Hacker's Delight §2.9

Reply:
Hmmm, I can see how this would work in some languages but go isn't as generous with the boolean arithmetic.

In python:
```python
>>> x, y = 1, 2
>>> (x > y) - (x < y)
-1
```
 vs. Go:
 ```go
x, y := 1, 2
(x > y) - (x < y)
	
`operator - not defined on (x > y) (untyped bool value)`
 ```
 
 Any suggestions for how this should be done in go? We could do some weird mapping / casting of bool to int but I think that overall the current if/else implementation would be cleaner.

Reply:
> Hmmm, I can see how this would work in some languages but go isn't as generous with the boolean arithmetic.

Quite right; never mind.